var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -63,6 +67,17 @@ class Zlib(MakefilePackage, Package):
 
     license("Zlib")
 
+    @classmethod
+    def determine_version(cls, lib):
+        match = re.search(r"lib\S\.so\.(\d+)\.(\d+)\.(\d+)", lib)
```

Comment:
This is a very strict regex, it precludes the matching of anything beyond a versioned, shared distribution on the Linux platform.
Please see the open-cv package's approach to this that is more platform/library type agnostic.
For Windows, I'll have to think about ways to get a version from a DLL, don't let that support hold up this PR.

Reply:
@johnwparent The regex was strict because it was only looking for `libz.so.x.x.x`  It won't work with all of the other libraries that you have suggested.  Can you please update the PR to have a robust regex for this then.

Reply:
```suggestion
for ext in library_extensions:
    if ext == "dylib":
        pattern = re.compile(fr"lib\S\.(\d+\.\d+\.\d+)\.{ext}")
    else:
        pattern = re.compile(fr"lib\S\.{ext}\.(\d+\.\d+\.\d+)")
    match = re.search(pattern, lib)
    if match:
        return match.group(1)
```

Reply:
That takes care of shared libraries on MacOS at the very least

Reply:
@johnwparent I actually think that the grouping parens on the patch are wrong.  I just merged it, but think that it needs to be corrected.